packages/drivers/odsp-driver/src/createFile.ts

```diff
@@ -37,8 +38,14 @@ const isInvalidFileName = (fileName: string): boolean => {
     return !!fileName.match(invalidCharsRegex);
 };
 
+// This map represents mapping from container package name to file extension.
+// For the packages that are not in this list, we will use .fluid extension.
+const fileExtensionMap = new Map([
```

Comment:
I do not think having that info in this repo is right thing. Why the world needs to know about onenote???
The right thing is to pass extension as part of newFileInfo

Reply:
The world doesn't care about onenote, but _ODSP_ does since this is an odsp convention that odsp is using to specify specific file metadata from fluid quorum metadata. The alternative is to have code in every host that parses the IFluidCodeDetails and has onenote specific container/odsp logic inside of all of them which will make host code much more odsp and onenote specific than we'd like (and would be a headache if odsp decided to do something like change file extensions/supported containers etc.). Also taking in extension info from the host opens the door for the host to misunderstand the odsp contract and potentially give very bad values to this driver for file extensions that odsp does not recognize and process as fluid, like .docx

Reply:
What is the scenario?
As I understand it, A host (let's say OneNote) wants to create a file and give it right extension.  Why other host (like Teams) needs to know about that? What if host1 decides to use 2, or 3 different extensions for their files? will we update driver and ask all other host to pick it up? Why?

I think host should be explicit, and pass extension it wants file to have.
The fact that it needs to be in sync with default ODSP logic on what file to invoke is not interesting to SPO driver - it's not its business to know about either of those things.

Reply:
Unfortunately the scenario is a lot more complex than that and does require multiple hosts :) The scenario is that X number of entrypoints (today this includes all Outlook Calendar apps and Teams) that create this particular container on the SPO driver should always generate a file of this extension type. In fact, OneNote is one of the few apps _not_ creating new files of this extension type. So we would need to build this logic into outlook and teams and not onenote :)

Reply:
The same scenario would extend to multiple containers supported by new filetypes in sharepoint, including whiteboard which the filetype is actually already known for. When _any_ host attaches a file against the SPO driver where the container is of a certain type, they expect the file type to be .whiteboard. Unfortunately, it's not host specific logic to create files in different locations or not, it's actually SPO Fluid policy based purely on quorum.

Reply:
Can you please provide more details on scenario?
Why this logic can't be in OWH layer?
This seems O365 specific logic, and that should not be in Fluid Framework repo, it should be in FCL repo whenever it happens.

Reply:
If the codeDetail is passed it, why we need to know about them?  It does seems like this should be in the OWH layer, which have specific knowledge of everything that it needs to support, including the rules about those extensions.

Reply:
Yeah, I think it could live in the OWH layer (as a helper exposed to hosts when calling create new APIs) but we try to keep driver specific logic out of office-web-host as much as possible. This would be a highly specific  SPO piece of logic exposed via office-web-host that would basically say "if you're using SPO, then call this helper around this SPO request to generate extra SPO details". Still, since it's in the SPO create new path anyways which means hosts already are in an SPO specific part of their logic, we can probably add it through office-web-host.

Reply:
I'd separate concerns:
1. SPO service knows about extension -> app mapping. It knows nothing about code proposal / package.
2. Some other layer need to know package -> extension mapping.

# 2 does not have to be SPO specific. Neither it needs to be constant. Same two hosts may have same experience (package), but use different extensions to isolate where content belongs.

I think # 2 should not be part of anything, it's a separate layer that all hosts may choice to use to ensure uniformity, if that's the goal. They may break that (stop using it) the moment they want to create special handling in this area.

Reply:
I don't think it's as simple as being a host decision for extensions since SPO has to specifically do work to enable extensions and app specific behavior related to those extensions. So SPO has quite a large say in these extensions and their behaviors whereas hosts just want to have a "consistent experience" for their Fluid experiences.

That said, we have a office-web-host helper layer that can explicitly be used for these consistent experiences and #2 may live there... but it will always be strongly SPO specific.

Reply:
So I think we will just expect a filename with proper extension from the host. If not supplied then we will fallback to .fluid extension.